packages/@aws-cdk/aws-codedeploy-api/lib/load-balancer.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * The generations of AWS load balancing solutions.
+ */
+export enum LoadBalancerGeneration {
+    /**
+     * The first generation (ELB Classic).
+     */
+    FIRST,
+
+    /**
+     * The second generation (ALB and NLB).
+     */
+    SECOND
+}
+
+/**
+ * An interface of an abstract laod balancer, as needed by CodeDeploy.
+ */
+export interface LoadBalancerProvider {
```

Comment:
Here's how we usually formulate such interfaces:

```ts
interface ILoadBalancer {
  readonly asCodeDeployLoadBalancer: LoadBalancerProps
}

interface LoadBalancerProps {
  generation: LoadBalancerGeneration;
  name: string;
}
```

Then, the implementors will look like this:

```ts
class ClassicLoadBalancer implements codedeployapi.ILoadBalancer {
  public get asCodeDeployLoadBalancer(): codedeployapi.LoadBalancerProps {
    return {
      generation: ...,
      name: this.blaName
   }
 }
}
```

The `asXxx` pattern signals the user that they shouldn't need to use this property directly. It's "a cast" to the specific interface.

Reply:
Done.